backend/services/userProfile.service.js

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable func-names */
+const mongoose = require('mongoose');
+const UserProfile = require('../models/userProfile.model');
+const modificationLogService = require('./modificationLog.service');
+
+const UserProfileService = {};
+
+UserProfileService.createUser = async function (userProfileData, authorEmail) {
+    
+    let newRecord = null;
+
+    const inputData = userProfileData;
+
+    if(Object.prototype.hasOwnProperty.call(inputData, "signupEmail") &&
+        inputData.signupEmail) {
+        inputData.signupEmail = userProfileData.signupEmail.toLowerCase();
+    }
+
+    await mongoose.connection.transaction(async () => {
+    
+        newRecord = await UserProfile.create(inputData);
+
+        await modificationLogService.saveLog(authorEmail, 
+            newRecord._id, "UserProfile", newRecord );          
+    });
+
+    return newRecord;
+    
+}
+
+
+UserProfileService.updateUser = async function (signupEmail, userProfileData, authorEmail) {
```

Comment:
Let's build this function in a way that does not allow to change the email by its design.
1. Use only `userProfileData, authorEmail` parameters
2. In this case, an updated email will mean that the mongoose doesn't find the user. So the email updating will be impossible.
3. We need to handle cases when an attempt to modify an object fails due to the absence of the object in the DB. I prefer to throw an error for this. We don't need to throw an error when we handle any `get` requests because we can just return null. But modifying should signalize that something goes wrong when a request tries to modify a non-existing object.

Reply:
Please look at new code.

Reply:
New Code throws new custom DatabaseError if object was not found.